controllers/operator_controller_test.go

```diff
@@ -293,10 +300,239 @@ var _ = Describe("Reconcile Test", func() {
+				Expect(err).NotTo(HaveOccurred())
+
+				apimeta.SetStatusCondition(&bd.Status.Conditions, metav1.Condition{
+					Type:    rukpakv1alpha1.TypeHasValidBundle,
+					Status:  metav1.ConditionUnknown,
+					Message: "unpacking",
+					Reason:  rukpakv1alpha1.ReasonUnpackSuccessful,
+				})
+
+				apimeta.SetStatusCondition(&bd.Status.Conditions, metav1.Condition{
+					Type:    rukpakv1alpha1.TypeInstalled,
+					Status:  metav1.ConditionUnknown,
+					Message: "installing",
+					Reason:  rukpakv1alpha1.ReasonInstallationSucceeded,
+				})
+
+				By("updating the status of bundleDeployment")
+				err = cl.Status().Update(ctx, &bd)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("running reconcile")
+				res, err := reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: opKey})
+				Expect(res).To(Equal(ctrl.Result{}))
+				Expect(err).NotTo(HaveOccurred())
+
+				By("fetching the updated operator after reconcile")
+				op := &operatorsv1alpha1.Operator{}
+				err = cl.Get(ctx, opKey, op)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("checking the expected conditions")
+				cond := apimeta.FindStatusCondition(op.Status.Conditions, operatorsv1alpha1.TypeReady)
+				Expect(cond).NotTo(BeNil())
+				Expect(cond.Status).To(Equal(metav1.ConditionUnknown))
+				Expect(cond.Reason).To(Equal(operatorsv1alpha1.ReasonInstallationSucceeded))
+				Expect(cond.Message).To(ContainSubstring(`waiting for rukpak to install bundleDeployment successfully`))
+			})
+
+			It("verify operator status when bundleDeployment installation status is unknown", func() {
+				err := cl.Get(ctx, opKey, &bd)
+				Expect(err).NotTo(HaveOccurred())
+
+				apimeta.SetStatusCondition(&bd.Status.Conditions, metav1.Condition{
+					Type:    rukpakv1alpha1.TypeInstalled,
+					Status:  metav1.ConditionUnknown,
+					Message: "installing",
+					Reason:  rukpakv1alpha1.ReasonInstallationSucceeded,
+				})
+
+				By("updating the status of bundleDeployment")
+				err = cl.Status().Update(ctx, &bd)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("running reconcile")
+				res, err := reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: opKey})
+				Expect(res).To(Equal(ctrl.Result{}))
+				Expect(err).NotTo(HaveOccurred())
+
+				By("fetching the updated operator after reconcile")
+				op := &operatorsv1alpha1.Operator{}
+				err = cl.Get(ctx, opKey, op)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("checking the expected conditions")
+				cond := apimeta.FindStatusCondition(op.Status.Conditions, operatorsv1alpha1.TypeReady)
+				Expect(cond).NotTo(BeNil())
+				Expect(cond.Status).To(Equal(metav1.ConditionUnknown))
+				Expect(cond.Reason).To(Equal(operatorsv1alpha1.ReasonInstallationSucceeded))
+				Expect(cond.Message).To(ContainSubstring(`waiting for rukpak to install bundleDeployment successfully`))
+			})
+		})
 		AfterEach(func() {
 			verifyInvariants(ctx, operator)
 
-			err := cl.Delete(ctx, operator)
+			err := cl.DeleteAllOf(ctx, operator)
```

Comment:
Curious about this change. Does DeleteAllOf do something different?

One thing I realize... We probably need to be cleaning up the BD. Garbage collection doesn't run in envtest.

Reply:
In one of the tests where the creation of duplicate package is tested, we create another operator object (https://github.com/operator-framework/operator-controller/blob/520219c1c4812d3446c85cf8383de3572ebbbf9d/controllers/operator_controller_test.go#L267)  which is left behind, that is why modified it to be `DeleteAllOf` to be on the safer side of cleaning up all operator objects.

That's a good point, though the owner references are set, we would still to delete BD, will modify the PR.

Reply:
Ah. In that case, we should have a separate AfterEach (at the same level as that BeforeEach) that deletes that other Operator.